pkg/netconfig/netconfig.go

```diff
@@ -208,6 +229,18 @@ func UpdateIPSet(ipVersion, ipSetName, egressFilterList, defaultNetworkDevice st
 	return nil
 }
 
+func RemoveIPSet(ipSetName string) error {
+	if err := DefaultNetUtilsCommandExecutor.ExecuteIPSetCommand("list", ipSetName); err == nil {
+		err = DefaultNetUtilsCommandExecutor.ExecuteIPSetCommand("destroy", ipSetName)
+		if err != nil {
+			return fmt.Errorf("error cleaning-up ipset %s: %w\n", ipSetName, err)
+		}
+	}
+
+	fmt.Printf("Removed ipset %s\n", ipSetName)
+	return nil
```

Comment:
Why not use the default branch-on-error way usual to go:

```suggestion
	if err := DefaultNetUtilsCommandExecutor.ExecuteIPSetCommand("list", ipSetName); err != nil {
		// ipset not found
		return nil
	}

	if err := DefaultNetUtilsCommandExecutor.ExecuteIPSetCommand("destroy", ipSetName); err != nil {
		return fmt.Errorf("error cleaning-up ipset %s: %w\n", ipSetName, err)
	}

	fmt.Printf("Removed ipset %s\n", ipSetName)
	return nil
```

Small side-effect of this would be that we do not print some misleading log that we had removed the ipset even when we did not.

Reply:
yes we could do that. I didn't want to change the existing patterns I saw in the code, e.g. [InitIPSet](https://github.com/gardener/egress-filter-refresher/blob/master/pkg/netconfig/netconfig.go#L124)

I didn't want to refactor the entire code (there are also other things missing, like proper logs for example) in the context of this story. 
But I agree there should only be logs for things that are actually done.